rest/database_init_manager.go

```diff
@@ -250,6 +278,9 @@ func (w *DatabaseInitWorker) Run() {
 		keyspaceCtx := base.KeyspaceLogCtx(w.ctx, w.n1qlStore.BucketName(), scName.ScopeName(), scName.CollectionName())
 		indexErr = db.InitializeIndexes(keyspaceCtx, w.n1qlStore, collectionIndexOptions)
 		if indexErr != nil {
+			if w.collectionStatusCallback != nil {
+				w.collectionStatusCallback(w.dbName, scName, CollectionIndexStatusError)
+			}
 			break
 		}
 
```

Comment:
The code below here is weird, because it actually should look like:

```
		// Check for context cancellation after each collection is processed - if cancelled, return cancellation error
		// to all watchers end exit
		if w.ctx.Err() != nil {
			indexErr = errors.New("Database initialization cancelled")
			break
        }
```

We check for `indexErr` in the case of a non ctx error above.